compiler/include/AstDump.h

```diff
@@ -116,6 +117,7 @@ class AstDump : public AstLogger {
   FILE*            mFP;             // The FILE* to the log file if the file is open
   int              mIndent;         // The indentation level.  Increments for each BlockStmt
   bool             mNeedSpace;      // Control inter-token spaces
+  bool             mDontCloseFP;    // true if mFP is owned by the user
```

Comment:
More than you wanted to take on, but it would seem that an appropriately-wrapped file object would eliminate the need for this crufty Boolean.  For example std::ostream problem knows that it should never close stdout....
Next time, right?